### ISSUE TITLE
Leading zeros stripped from string keys that cast to integers.

### ISSUE THREAD

**Issue by trxcllnt** [Netflix/falcor-router#68]
@michaelbpaulson I understand we're casting strings to ints, and it would seem this key could be a candidate for such a cast, but since my route is matching `Router.keys`, shouldn't the matcher skip the cast? I suppose if we _have_ to cast, we should verify the new integer's digit count is the same as the original string.

```javascript
var router = new Router([{
    route: 'lolomo',
    get: function(path) {
        return Rx.Observable.return({
            paths: [path],
            jsonGraph: {
                lolomo: $ref(["lolomos", "0123456789"]) // <-- bug part 1
            }
        })
        .do(function(x) {
            console.log(JSON.stringify(x));
        });
    }
}, {
    route: 'lolomos[{keys:lolomoID}][{keys:lists}]',
    get: function(path) {
        var lolomoID = path.lolomoID[0];
        console.log(lolomoID) // <-- bug part 2: should print String "0123456789" -- actually prints Number 123456789
        return Rx.Observable
            .from(path.lists)
            .reduce(function(lists, i) {
                lists[i] = $ref(["lists", i]);
                return lists;
            }, {})
            .map(function(lists) {
                var lolomos = {};
                lolomos[lolomoID] = lists;
                return {
                    paths: [path],
                    jsonGraph: { lolomos: lolomos }
                };
            })
            .do(function(x) {
                console.log(JSON.stringify(x));
            });
    }
}]);
```

**Comment by jhusain** [comment]
"{keys}" should not cast to number.

**Comment by lrowe** [comment]
Fixed in falcor-path-utils 0.5.1 by Netflix/falcor-path-utils#11.

**Comment by alexmuro** [comment]
when installing falcor-router for a server side project npm installs falcor-path-utils 0.5.0 as a dependency and I had to manually upgrade to 0.5.1 to fix this bug. Not a huge deal all in all but doesn't work out of the box.
